drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -830,38 +831,137 @@ local function do_bridge_network_init(driver, device, bridge_url, api_key)
                     string.format(
                       "Light device \"%s\" was deleted from hue bridge %s",
                       (light_device.label or light_device.id or "unknown device"),
-                      (device.label or device.device_network_id or device.id or "unknown bridge")
+                      (bridge_device.label or bridge_device.device_network_id or bridge_device.id or "unknown bridge")
                     )
                   )
-                  light_device:set_field(Fields.IS_ONLINE, false)
-                  light_device:offline()
+                  light_device.log.trace("Attempting to delete Device UUID " .. tostring(light_device.id))
+                  driver:do_hue_light_delete(light_device)
                 end
               end
             end
           elseif event.type == "add" then
             for _, add_data in ipairs(event.data) do
-              if add_data.type == "light" then
+              if add_data.type == "light" and add_data.owner and add_data.owner.rtype == "device" then
                 log.info(
                   string.format(
-                    "New light added to Hue Bridge \"%s\", light properties: \"%s\", " ..
-                    "re-run discovery to join new lights to SmartThings",
-                    device.label, st_utils.stringify_table(add_data, nil, false)
+                    "New light added to Hue Bridge \"%s\", light properties: \"%s\"",
+                    bridge_device.label, json.encode(add_data)
                   )
                 )
+
+                cosock.spawn(function()
```

Comment:
This code seems similar to what exists in `disco.lua`. Could it be combined?

Reply:
Yup; in fact, there's a *ton* of code that's not very `DRY` in this driver.

I'd like to make a separate PR that is a purely hygeine refactor without any behavior changes to catch some of this stuff.